simpleclient_hibernate/src/main/java/io/prometheus/client/hibernate/HibernateStatisticsCollector.java

```diff
@@ -0,0 +1,472 @@
+package io.prometheus.client.hibernate;
+
+import io.prometheus.client.Collector;
+import io.prometheus.client.CollectorRegistry;
+import io.prometheus.client.CounterMetricFamily;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.hibernate.SessionFactory;
+import org.hibernate.stat.Statistics;
+
+/**
+ * Collect metrics from one or more Hibernate SessionFactory instances.
+ * <p>
+ * Usage example for a single session factory:
+ * <pre>
+ * new HibernateStatisticsCollector(sessionFactory, "myapp").register();
+ * </pre>
+ * Usage example for multiple session factories:
+ * <pre>
+ * new HibernateStatisticsCollector()
+ *     .add(sessionFactory1, "myapp1")
+ *     .add(sessionFactory2, "myapp2")
+ *     .register();
+ * </pre>
+ * If you are in a JPA environment, you can obtain the SessionFactory like this:
+ * <pre>
+ * SessionFactory sessionFactory =
+ *     entityManagerFactory.unwrap(SessionFactory.class);
+ * </pre>
+ *
+ * @author Christian Kaltepoth
+ */
+public class HibernateStatisticsCollector extends Collector {
+
+  private static final List<String> LABEL_NAMES = Collections.singletonList("unit");
+
+  private final Map<String, SessionFactory> sessionFactories = new LinkedHashMap<String, SessionFactory>();
```

Comment:
Hmm, let's make this a ConcurrentHashMap so that different threads can add themselves without having to worry about synchronisation.